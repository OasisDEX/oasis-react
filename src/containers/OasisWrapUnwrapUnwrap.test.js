/* global shallow describe it expect */
/* eslint-disable import/first */
import React from 'react';
import { fromJS } from 'immutable';

import {
  OasisWrapUnwrapUnwrapWrapper,
  mapStateToProps,
  mapDispatchToProps
} from './OasisWrapUnwrapUnwrap';
import { shallow } from 'enzyme';

describe('(Container) OasisWrapUnwrapUnwrap', () => {
  const state = fromJS(global.storeMock);
  const initialProps = mapStateToProps(state);
  const initialActions = mapDispatchToProps(x => x);
  const props = {
    ...initialActions,
    ...initialProps
  };

  it('will receive right props', () => {
    expect(initialProps).toMatchSnapshot();
  });


  it('will receive right actions', () => {
    expect(initialActions).toMatchSnapshot();
  });

  it('should render', () => {
    const wrapper = shallow(
      <OasisWrapUnwrapUnwrapWrapper {...props}/>
    );
    expect(wrapper).toMatchSnapshot();
  });

});
