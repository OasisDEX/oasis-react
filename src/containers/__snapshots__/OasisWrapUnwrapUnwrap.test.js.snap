// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`(Container) OasisWrapUnwrapUnwrap should render 1`] = `
<OasisWrapUnwrapUnwrap
  activeWrappedToken="W-GNT"
  activeWrappedTokenBalance={"0"}
  onSubmit={[Function]}
/>
`;

exports[`(Container) OasisWrapUnwrapUnwrap will receive right actions 1`] = `
Object {
  "actions": Object {
    "unwrapToken": [Function],
  },
}
`;

exports[`(Container) OasisWrapUnwrapUnwrap will receive right props 1`] = `
Object {
  "activeWrappedToken": "W-GNT",
  "activeWrappedTokenBalance": "0",
}
`;
